### PR TITLE
workloads: Added opea-chatqna workload yaml's & readme

### PR DESCRIPTION
**Pull request by vbedida79**
Added opea-chatqna workload yaml's for rhoai/setup on ocp and readme.
Updated e2e/inference/readme with Intel Gaudi accelerator profile details

Signed-off-by: vbedida79 <veenadhari.bedida@intel.com>

### CODE REVIEW

File: e2e/inference/README.md [hunk 1]
```diff
@@ -64,5 +64,8 @@ Follow the [link](https://github.com/openvinotoolkit/operator/blob/main/docs/not
 ![Alt text](/docs/images/Ovms-Gpu-resource-request.png)
 
 
+# Intel Gaudi Accelerator on RHOAI
```
Comment (uMartinXu) [inline e2e/inference/README.md:67]: # Enable Intel Gaudi AI Accelerator with RHOAI

File: e2e/inference/README.md [hunk 2]
```diff
@@ -64,5 +64,8 @@ Follow the [link](https://github.com/openvinotoolkit/operator/blob/main/docs/not
...
+To enable and use the Intel Gaudi accelerator on RHOAI, follow the [documentation](https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2.11/html/working_with_accelerators/intel-gaudi-ai-accelerator-integration_accelerators#enabling-intel-gaudi-ai-accelerators_accelerators).
+
```
Comment (uMartinXu) [inline e2e/inference/README.md:69]: The doc here contains the instructions for creating the Accelerator profile from WebConsole manually. I think we can provide a reference yaml file for the user to create the Accelerator profile automatically. And we can also integrate this into the one-click solution later.

File: security/anyuid_rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+metadata:
+  name: opea-chatqna
+  namespace: opea-chatqna
+rules:
+- apiGroups:
+  - security.openshift.io
+  resources:
+  - securitycontextconstraints
+  resourceNames:
+  - anyuid
```
Comment (uMartinXu) [inline security/anyuid_rbac.yaml:21]: @vbedida79  For the prototype release, I am ok to use anyuid. However, we should figure out the reason why "anyuid" needs to be used. Could we file an issue to track the task? Thanks!

File: e2e/inference/README.md [hunk 4]
```diff
@@ -39,7 +39,7 @@ Create `AcceleratorProfile` in the `redhat-ods-applications` namespace
 $ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/e2e/inference/accelerator_profile.yaml
```
Comment (uMartinXu) [inline e2e/inference/README.md:39]: @vbedida79  Could we split the e2e related contents to another PR?

File: workloads/opea/chatqna/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+```
+* Check persistent volume claims are bound to the volumes:
+```
+NAME                          STATUS   VOLUME                       CAPACITY   ACCESS MODES   
+chatqna-megaservice-pvc-0   Bound    chatqna-megaservice-pv-0   100Mi      RWO            
+chatqna-megaservice-pvc-1   Bound    chatqna-megaservice-pv-1   100Mi      RWO            
+chatqna-megaservice-pvc-2   Bound    chatqna-megaservice-pv-2   100Mi      RWO            
+```
+2. From Webconsole
```
Comment (hershpa) [inline workloads/opea/chatqna/README.md:35]:
```suggestion
2. From Web Console
```

File: workloads/opea/chatqna/README.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+### Text Generation Inference for Intel Gaudi on RHOAI
+1. Access ```rhods-dashboard``` link from routes on your web console and click on Data Science Projects. Now, create a project.
+
+2. Click on the project, and choose ```Data Connection``` section. In the fields, add your access and secret keys from minio follow [prerequisites section](#prerequisites). This downloads the model and prepares for serving.
+
+3. Now click on the Settings and choose ```ServingRuntime```. Copy or import the [tgi_gaudi_servingruntime.yaml](../tgi-gaudi-servingruntime.yaml). The [tgi-gaudi](https://github.com/huggingface/tgi-gaudi) serving runtime is used.
+
+4. Go back to your project and choose ```Models```. Choose the servingruntime and model from the dropdown (from step 3). Choose the number of accelerators (from step 3), size of container and the data connection- minio (created from step 2). Add the appropriate folder in your bucket for the data connection.
+
+5. Now the model server is in the creation state. Once ready, the status will be updated to green and the inference endpoint can be seen. Update this inference endpoint in the backend deployment and restart it to run inference requests.
```
Comment (hershpa) [inline workloads/opea/chatqna/README.md:78]:
```suggestion
5. The model server is now in the creation state. Once ready, the status will be updated to green and the inference endpoint can be seen. Update this inference endpoint in the backend deployment and restart it to run inference requests.
```

File: e2e/inference/README.md (hunk 2 above)
Comment (vbedida79) [reply e2e/inference/README.md]: Updated in PR https://github.com/intel/intel-technology-enabling-for-openshift/pull/285

File: e2e/inference/README.md (hunk 4 above)
Comment (vbedida79) [reply e2e/inference/README.md]: Done. Added in https://github.com/intel/intel-technology-enabling-for-openshift/pull/285

File: workloads/opea/chatqna/README.md (hunk 5 above)
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: updated, ty.

File: workloads/opea/chatqna/README.md (hunk 6 above)
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: updated, thanks

**Review by hershpa — APPROVED** [review]
LGTM

File: workloads/opea/chatqna/README.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
+# OPEA ChatQnA workload on OCP
+
+## Overview
+The workload is based on the [OPEA ChatQnA](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA) running on Intel® Gaudi Accelerator with OpenShift and OpenShift AI. Refer to the OPEA [readme](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/README.md) for more details about the workload.
```
[the remark this answers is not in the document]
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: updated

File: workloads/opea/chatqna/README.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
+# OPEA ChatQnA workload on OCP
```
[the remark this answers is not in the document]
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: updated, ty.

File: workloads/opea/chatqna/README.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+## Overview
+The workload is based on the [OPEA ChatQnA](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA) running on Intel® Gaudi Accelerator with OpenShift and OpenShift AI. Refer to the OPEA [readme](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/README.md) for more details about the workload. 
+ 
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent Storage such as NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html)
```
[the remark this answers is not in the document]
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: done, thanks

File: workloads/opea/chatqna/README.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
+# Deploy OPEA ChatQnA workload on OCP
+
+## Overview
+The workload is based on the [OPEA ChatQnA Application](https://github.com/opea-project/GenAIExamples/tree/v0.8/ChatQnA) running on Intel® Gaudi Accelerator with OpenShift and OpenShift AI. Refer to the OPEA [OPEA Generative AI Examples](https://github.com/opea-project/GenAIExamples/tree/v0.8) for more details about the OPEA workloads.
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:4]: OPEA [OPEA Generative AI Examples] ->[OPEA Generative AI Examples]

File: workloads/opea/chatqna/README.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
+# Deploy OPEA ChatQnA workload on OCP
+
+## Overview
+The workload is based on the [OPEA ChatQnA Application](https://github.com/opea-project/GenAIExamples/tree/v0.8/ChatQnA) running on Intel® Gaudi Accelerator with OpenShift and OpenShift AI. Refer to the OPEA [OPEA Generative AI Examples](https://github.com/opea-project/GenAIExamples/tree/v0.8) for more details about the OPEA workloads. 
+ 
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent storage using NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/persistent_storage/persistent-storage-nfs.html) for the details to set it up.
+
+**Note**: Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html) for setting up other persistent storages.
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:10]: for setting up other type of persistent storages

File: workloads/opea/chatqna/README.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
+# OPEA ChatQnA workload on OCP
+
+## Overview
+The workload is based on the [OPEA ChatQnA](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA) running on Intel® Gaudi Accelerator with OpenShift and OpenShift AI. Refer to the OPEA [readme](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/README.md) for more details about the workload. 
+ 
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent Storage such as NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html)
+* Provisioned Intel Gaudi accelerator on RHOCP cluster. Follow steps [../../../gaudi/README.md]
+* Installed and Configured RHOAI. Follow steps [here](../inference/README.md/#install-rhoai)
```
[the remark this answers is not in the document]
Comment (uMartinXu) [reply workloads/opea/chatqna/README.md:12]: Need to update the link after PR285 got merged.

File: workloads/opea/chatqna/README.md (hunk 11 above)
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: done, thanks

File: workloads/opea/chatqna/README.md (hunk 10 above)
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: updated

File: workloads/opea/chatqna/README.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+ 
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent storage using NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/persistent_storage/persistent-storage-nfs.html) for the details to set it up.
+
+**Note**: Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html) for setting up other types of persistent storages.
+* Provisioned Intel Gaudi accelerator on RHOCP cluster. Follow steps [../../../gaudi/README.md]
+* Installed and Configured RHOAI. Follow steps [here](../inference/README.md/#install-rhoai) 
+* Minio based S3 service ready for RHOAI. Follow steps [here](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio)
+* Create the Persistent Volume for the Application
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:14]: ## Create Persistent Volumes
The NFS is used to create the Persistent Volumes for ChatQnA MegaService to claim and use.
Make sure to update NFS server IP and path in  ```persistent_volumes.yaml``` before applying command below. 
For example:
```
  nfs:
    server: 192.168.0.21 # nfs server
    path: /my_nfs # nfs path
```

``` 
$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/persistent_volumes.yaml
```

File: workloads/opea/chatqna/README.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+Please update the NFS server IP and the path in the ```persistent_volumes.yaml``` file and deploy:
+
+``` 
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/persistent_volumes.yaml
+
+```
+``` 
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/persistent_volume_claims.yaml
+
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:38]: Can we merge the persistent_volumes_claims.yaml with chatqna_megaservice_deploy.yaml, and remove the steps to claim the persistent volumes?

File: workloads/opea/chatqna/README.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+```
+* Check persistent volume claims are bound to the volumes:
+```
+NAME                          STATUS   VOLUME                       CAPACITY   ACCESS MODES   
+chatqna-megaservice-pvc-0   Bound    chatqna-megaservice-pv-0   100Mi      RWO            
+chatqna-megaservice-pvc-1   Bound    chatqna-megaservice-pv-1   100Mi      RWO            
+chatqna-megaservice-pvc-2   Bound    chatqna-megaservice-pv-2   100Mi      RWO            
+```
+2. From Web Console
+
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:48]: Let's make it simple and remove the "From Web Console" step here

File: workloads/opea/chatqna/README.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent storage using NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/persistent_storage/persistent-storage-nfs.html) for the details to set it up.
+
+**Note**: Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html) for setting up other types of persistent storages.
+* Provisioned Intel Gaudi accelerator on RHOCP cluster. Follow steps [../../../gaudi/README.md]
+* Installed and Configured RHOAI. Follow steps [here](../inference/README.md/#install-rhoai) 
+* Minio based S3 service ready for RHOAI. Follow steps [here](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio)
+* Create the Persistent Volume for the Application
+First, create ```oc create namespace opea-chatqna``` namespace.
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md]: ## Create Namespace and Setup Certificate
Use the below command to create name space **opea-chatqna**
```
oc create namespace opea-chatqna
```

Use the command below to set up the Certificate for ChatQnA MegaService to access the backend ChatQnA Microservices HTTPS endpoint
```
./setup_certificate.sh opea-chatqna
```

File: workloads/opea/chatqna/README.md (hunk 13 above)
Comment (vbedida79) [reply workloads/opea/chatqna/README.md]: > ## Create Persistent Volumes
> The NFS is used to create the Persistent Volumes for ChatQnA MegaService to claim and use. Make sure to update NFS server IP and path in `persistent_volumes.yaml` before applying command below. For example:
> 
> ```
>   nfs:
>     server: 192.168.0.21 # nfs server
>     path: /my_nfs # nfs path
> ```
> 
Updated
> @vbedida79 please add the command to verify the Persistent Volumes
the command below  shows both persistent volumes and persistent volumes created and bound. this verifies both.

File: workloads/opea/chatqna/setup_certificate.sh [hunk 17]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024 Intel Corporation
```
Comment (uMartinXu) [inline workloads/opea/chatqna/setup_certificate.sh:1]: Can we change it to setup_namespace_certificate.sh

File: workloads/opea/chatqna/setup_certificate.sh [hunk 18]
```diff
@@ -0,0 +1,6 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#!/bin/bash
+NAMESPACE="opea-chatqna"
```
Comment (uMartinXu) [inline workloads/opea/chatqna/setup_certificate.sh:4]: oc create namespace opea-chatqna

File: workloads/opea/chatqna/README.md [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent storage using NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/persistent_storage/persistent-storage-nfs.html) for the details to set it up.
+
+**Note**: Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html) for setting up other types of persistent storages.
+* Provisioned Intel Gaudi accelerator on RHOCP cluster. Follow steps [../../../gaudi/README.md]
+* Installed and Configured RHOAI. Follow steps [here](../inference/README.md/#install-rhoai) 
+* Minio based S3 service ready for RHOAI. Follow steps [here](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio)
+* Create Persistent Volumes
+
+First, create namespace: ```oc create namespace opea-chatqna```.
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:16]: remove line 15, 16

File: workloads/opea/chatqna/README.md [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+ 
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent storage using NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/persistent_storage/persistent-storage-nfs.html) for the details to set it up.
+
+**Note**: Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html) for setting up other types of persistent storages.
+* Provisioned Intel Gaudi accelerator on RHOCP cluster. Follow steps [../../../gaudi/README.md]
+* Installed and Configured RHOAI. Follow steps [here](../inference/README.md/#install-rhoai) 
+* Minio based S3 service ready for RHOAI. Follow steps [here](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio)
+* Create Persistent Volumes
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:14]: change line 14 with 
## Create Persistent Volumes

File: workloads/opea/chatqna/README.md [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+
+First, create namespace: ```oc create namespace opea-chatqna```.
+
+The NFS is used to create the Persistent Volumes for ChatQnA MegaService to claim and use.
+
+Make sure to update NFS server IP and path in ```persistent_volumes.yaml``` before applying command below.
+For example:
+```
+  nfs:
+    server: 0.0.0.0 # nfs server
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:24]: let's use a real IP as example not use the broadcast address 0.0.0.0.

File: workloads/opea/chatqna/README.md [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+    path: /my_nfs # nfs path
+```
+
+1. From CLI
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:28]: to keep it simple, lets only provide the CLI way, and remove the web console way.

File: workloads/opea/chatqna/README.md [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+  
+**Note**: The NFS is used to provide the Persistent Volume as reference here.
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:30]: remove line 30, since it is already mentioned above.

File: workloads/opea/chatqna/README.md [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+1. From CLI
+  
+**Note**: The NFS is used to provide the Persistent Volume as reference here. 
+
+``` 
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/persistent_volumes.yaml
+
+```
+``` 
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/persistent_volume_claims.yaml
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:37]: Could we remove it and add the command to verify the persistent volumes?

File: workloads/opea/chatqna/README.md [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+```
+2. From Web Console
+
+Navigate to ```Storage``` -> ```PersistentVolumes```. Click on ```CreatePersistentVolume``` and copy the above yaml. Repeat the same for ```PersistentVolumeClaims```.
+
+### ChatQnA backend- langchain and redis vector database
+1. Deploy anyuid_rbac required for redis vector db and chatqna megaservice
+
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/security/anyuid_rbac.yaml
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:55]: Can we merge this anyuid_rbac.yaml with redis_deployment_service.yaml and remove this step to make it simple?

File: workloads/opea/chatqna/README.md [hunk 26]
```diff
@@ -0,0 +1,101 @@
...
+```
+NAME                          STATUS   VOLUME                       CAPACITY   ACCESS MODES   
+chatqna-megaservice-pvc-0   Bound    chatqna-megaservice-pv-0   100Mi      RWO            
+chatqna-megaservice-pvc-1   Bound    chatqna-megaservice-pv-1   100Mi      RWO            
+chatqna-megaservice-pvc-2   Bound    chatqna-megaservice-pv-2   100Mi      RWO            
+```
+2. From Web Console
+
+Navigate to ```Storage``` -> ```PersistentVolumes```. Click on ```CreatePersistentVolume``` and copy the above yaml. Repeat the same for ```PersistentVolumeClaims```.
+
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:50]: Could we move the building of MegaService here?
## Building OPEA ChatQnA MegaService Container Image
The command below can be used to download the ChatQnA MegaService from [OPEA project](https://github.com/opea-project/GenAIExamples.git) and build the container image on-premise on OCP.

```
$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/chatqna_megaservice_buildconfig.yaml
```

File: workloads/opea/chatqna/README.md [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+3. Build the chatqna megaservice image
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/chatqna_megaservice_buildconfig.yaml
+
+```
+
+4. Setup the Certificate for accessing the https endpoint.
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:71]: this can be removed

File: workloads/opea/chatqna/README.md [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+NAME                          STATUS   VOLUME                       CAPACITY   ACCESS MODES   
+chatqna-megaservice-pvc-0   Bound    chatqna-megaservice-pv-0   100Mi      RWO            
+chatqna-megaservice-pvc-1   Bound    chatqna-megaservice-pv-1   100Mi      RWO            
+chatqna-megaservice-pvc-2   Bound    chatqna-megaservice-pv-2   100Mi      RWO            
+```
+2. From Web Console
+
+Navigate to ```Storage``` -> ```PersistentVolumes```. Click on ```CreatePersistentVolume``` and copy the above yaml. Repeat the same for ```PersistentVolumeClaims```.
+
+### ChatQnA backend- langchain and redis vector database
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:51]: remove line 51

File: workloads/opea/chatqna/README.md [hunk 29]
```diff
@@ -0,0 +1,101 @@
...
+
+```
+
+2. Deploy the redis vector database deployment and service
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/redis_deployment_service.yaml
+
+```
+
+3. Build the chatqna megaservice image
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:65]: this line can be removed

File: workloads/opea/chatqna/README.md [hunk 30]
```diff
@@ -0,0 +1,101 @@
...
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/chatqna_megaservice_buildconfig.yaml
+
+```
+
+4. Setup the Certificate for accessing the https endpoint. 
+  ```
+  ./setup_certificate.sh 
+  ```
+
+5. Deploy the chatqna-megaservice pod
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:76]: ## Deploy ChatQnA MegaService

File: workloads/opea/chatqna/README.md [hunk 31]
```diff
@@ -0,0 +1,101 @@
...
+
+### ChatQnA backend- langchain and redis vector database
+1. Deploy anyuid_rbac required for redis vector db and chatqna megaservice
+
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/security/anyuid_rbac.yaml
+
+```
+
+2. Deploy the redis vector database deployment and service
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:59]: ## Deploy Redis Vector Database Service

File: workloads/opea/chatqna/README.md [hunk 32]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+### Text Generation Inference for Intel Gaudi on RHOAI
+1. Access ```rhods-dashboard``` link from routes on your web console and click on Data Science Projects. Now, create a project.
+
+2. Click on the project, and choose ```Data Connection``` section. In the fields, add your access and secret keys from minio follow [prerequisites section](#prerequisites). This downloads the model and prepares for serving.
+
+3. Now click on the Settings and choose ```ServingRuntime```. Copy or import the [tgi_gaudi_servingruntime.yaml](../tgi-gaudi-servingruntime.yaml). The [tgi-gaudi](https://github.com/huggingface/tgi-gaudi) serving runtime is used.
+
+4. Go back to your project and choose ```Models```. Choose the servingruntime and model from the dropdown (from step 3). Choose the number of accelerators (from step 3), size of container and the data connection- minio (created from step 2). Add the appropriate folder in your bucket for the data connection.
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:88]: What Models will we use here? 
Will we need to download the Model manually or RHOAI will download the Model for us?

Comment (vbedida79) [reply workloads/opea/chatqna/README.md:88]: Models supported in opea readme. tried with llama2. we use minio setup for model storage (in step 2 data connection). rhoai model server downloads from minio

File: workloads/opea/chatqna/README.md (hunk 14 above)
Comment (vbedida79) [reply workloads/opea/chatqna/README.md:38]: persistent volume claims take some time to bind to the volumes. and deployment is deployed at the same time. if the binding takes a while deployment might error as well. Hence the separate file

**Comment by vbedida79** [comment]
@uMartinXu updated with the requested changes. Please review.

File: workloads/opea/chatqna/README.md [hunk 33]
```diff
@@ -0,0 +1,110 @@
+# Deploy OPEA ChatQnA workload on OCP
+
+## Overview
+The workload is based on the [OPEA ChatQnA Application](https://github.com/opea-project/GenAIExamples/tree/v0.8/ChatQnA) running on Intel® Gaudi Accelerator with OpenShift and OpenShift AI. Refer to the [OPEA Generative AI Examples](https://github.com/opea-project/GenAIExamples/tree/v0.8) for more details about the OPEA workloads. 
+ 
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent storage using NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/persistent_storage/persistent-storage-nfs.html) for the details to set it up.
```
[the remark this answers is not in the document]
Comment (vbedida79) [reply workloads/opea/chatqna/README.md:8]: I think this is ok. cluster has oc tool details as well, while and in the provisioning documentation from rh

File: workloads/opea/chatqna/README.md [hunk 34]
```diff
@@ -0,0 +1,110 @@
...
+
+```
+NAME                           CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      
+chatqna-megaservice-pv-0        100Mi      RWO            Retain           Available
+chatqna-megaservice-pv-1        100Mi      RWO            Retain           Available
+chatqna-megaservice-pv-2        100Mi      RWO            Retain           Available
+
+```
+### Building OPEA ChatQnA MegaService Container Image
+* Clone the [repo](https://github.com/opea-project/GenAIExamples). From the [folder](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA/deprecated/langchain/docker), run the following commands. Ensure that the certificate downloaded before is copied to this folder.
```
[the remark this answers is not in the document]
Comment (vbedida79) [reply workloads/opea/chatqna/README.md:53]: added as commands for now

File: workloads/opea/chatqna/README.md (hunk 33 above)
[the remark this answers is not in the document]
Comment (vbedida79) [reply workloads/opea/chatqna/README.md:10]: this is part of cluster when provisioning. so we need not state explicitly

**Comment by vbedida79** [comment]
updated based on latest review. ready for merge @uMartinXu

File: workloads/opea/chatqna/README.md [hunk 35]
```diff
@@ -0,0 +1,124 @@
...
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* The Persistent storage using NFS is ready. Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/persistent_storage/persistent-storage-nfs.html) for the details to set it up.
+
+    **Note**: Refer to [documentation](https://docs.openshift.com/container-platform/4.16/storage/index.html) for setting up other types of persistent storages.
+* Provisioned Intel Gaudi accelerator on RHOCP cluster. Follow steps [here](/gaudi/README.md)
+* RHOAI is installed. Follow steps [here](../inference/README.md/#install-rhoai) 
+* The Intel Gaudi AI accelerator is enabled with RHOAI. Follow steps [here]((../inference/README.md/#enable-intel-gaudi-ai-accelerator-with-rhoai))
+* Minio based S3 service ready for RHOAI. Follow steps [here](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio)
+### Create namespace and download certificate
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:17]: ### Create namespace

File: workloads/opea/chatqna/README.md [hunk 36]
```diff
@@ -0,0 +1,124 @@
...
+
+```
+    $ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/chatqna_megaservice_buildconfig.yaml
+
+```
+```
+    $ oc start-build chatqna-megaservice --from-dir=./ --follow
+
+```
+### ChatQnA backend- langchain and redis vector database
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:75]: remove this line 75 please.

File: workloads/opea/chatqna/README.md [hunk 37]
```diff
@@ -0,0 +1,124 @@
...
+
+#### Deploy Redis Vector Database Service
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/redis_deployment_service.yaml
+
+```
+
+Check that the pod is running:
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:83]: We need to check the service is running.

File: workloads/opea/chatqna/README.md [hunk 38]
```diff
@@ -0,0 +1,124 @@
...
+NAME                                   READY   STATUS      RESTARTS   AGE
+redis-vector-db-6b5747bf7-sl8fr        1/1     Running     0          20s
+```
+
+#### Deploy ChatQnA MegaService
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/chatqna_megaservice_deployment.yaml
+```
+
+Check that the pod is running:
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:96]: We need to check the service is running

File: workloads/opea/chatqna/README.md [hunk 39]
```diff
@@ -0,0 +1,132 @@
...
+
+* Follow the dashboard and click ```Data Science Projects``` to create a project. For example, ```OPEA-chatqna-modserving```.
+
+### Preload the models
+
+* Refer to [link](https://huggingface.co/docs/hub/en/models-downloading) and download the model [Llama2-70b-chat-hf](https://huggingface.co/meta-llama/Llama-2-70b-chat-hf). 
+
+* Refer to [link](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio) and upload the model to minio/s3 storage. 
+
+* Click ```OPEA-chatqna-modserving```, and choose ```Data Connection``` section. In the fields, add your access and secret keys from minio- follow [prerequisites section](#prerequisites).
```
Comment (uMartinXu) [inline workloads/opea/chatqna/README.md:32]: we need more specific link here

**Review by uMartinXu — APPROVED** [review]
